odoo/addons/base/models/res_users.py

```diff
@@ -2285,6 +2287,7 @@ class APIKeys(models.Model):
     name = fields.Char("Description", required=True, readonly=True)
     user_id = fields.Many2one('res.users', index=True, required=True, readonly=True, ondelete="cascade")
     scope = fields.Char("Scope", readonly=True)
+    expiration_date = fields.Datetime("Expiration Date", readonly=True)
```

Comment:
This field should be computed as `create_date + apikey_duration`.
Since the requirement is "they must have a date", make this required and remove IS NULL checks in SQL. If the administrator wishes, they can set a very high expiration date.